esphome/components/uart/event/uart_event.cpp

```diff
@@ -0,0 +1,48 @@
+#include "uart_event.h"
+#include "esphome/core/log.h"
+#include <algorithm>
+
+namespace esphome::uart {
+
+static const char *const TAG = "uart.event";
+
+void UARTEvent::setup() { buffer_.clear(); }
```

Comment:
```suggestion
void UARTEvent::setup() { this->buffer_.clear(); }
```

nit: https://developers.esphome.io/contributing/code/#c-code-style

Reply:
Actually I think we can drop this, the vector is already empty at setup time